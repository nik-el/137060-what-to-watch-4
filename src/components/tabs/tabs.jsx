import React from "react";
import PropTypes from 'prop-types';

export const Tabs = React.memo(function Tabs({tabs, active, onTabClick}) {
  const handleTabClick = (event, tab) => {
    event.preventDefault();
    onTabClick(tab);
  };

  const getTabClass = (tab) => {
    return `movie-nav__item` + (tab === active ? ` movie-nav__item--active` : ``);
  };

  return <ul className="movie-nav__list">
    { tabs.map((tab, i) => (
      <li key={tab + i} className={getTabClass(tab)}>
        <a href="#" onClick={(e) => handleTabClick(e, tab)} className="movie-nav__link">{tab}</a>
      </li>
    ))}
  </ul>;
});


Tabs.propTypes = {
  // список табов
  tabs: PropTypes.arrayOf(PropTypes.string),
  // активый таб
  active: PropTypes.string,
  // обработчик клика по табу
  onTabClick: PropTypes.func
};
