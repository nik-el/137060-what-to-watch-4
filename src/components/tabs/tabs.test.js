import React from 'react';
import renderer from 'react-test-renderer';
import {Tabs} from './tabs';
import {BrowserRouter as Router} from 'react-router-dom';

const TEST_TABS = [`tab 1`, `tab 2`];
const currentTab = TEST_TABS[0];

const onTabClick = () => {};

it(`Tabs renders correctly`, () => {
  const tree = renderer
    .create(
        <Router>
          <Tabs
            tabs={TEST_TABS}
            active={currentTab}
            onTabClick={onTabClick}
          />
        </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

