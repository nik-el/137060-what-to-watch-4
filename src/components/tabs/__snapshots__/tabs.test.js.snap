// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
<ul
  className="movie-nav__list"
>
  <li
    className="movie-nav__item movie-nav__item--active"
  >
    <a
      className="movie-nav__link"
      href="#"
      onClick={[Function]}
    >
      tab 1
    </a>
  </li>
  <li
    className="movie-nav__item"
  >
    <a
      className="movie-nav__link"
      href="#"
      onClick={[Function]}
    >
      tab 2
    </a>
  </li>
</ul>
`;
